straxen/corrections_services.py

```diff
@@ -131,13 +132,21 @@ def _get_correction(self, run_id, correction, version):
                 for it_correction in pmts: # loop over all PMTs
                     if correction in it_correction:
                         df = self.interface.read(it_correction)
+                        if self.check_for_nans(when, df, version):
```

Comment:
Im wondering if this is the correct place to check for nans. This essentially checks whether any nans exist before the requests time so I would get an error even if there are nans far back in the beginning of the science run that dont effect the run I requested right? maybe only check the period in time that effects the run? e.g after interpolation and selection

Reply:
I'm not sure we want a "version" with discontinuities(nans). 
If that is the case we will need to define a period for where to check for nans which is totally arbitrary. 
Doing any check after the interpolation does not make sense since DataFrame.Interpolate() fill nans.

Reply:
> If that is the case we will need to define a period for where to check for nans which is totally arbitrary.
Im not sure I follow, in the current API a correction is always requested for a specific run and a run defines a very specific time range so why is that arbitrary?

Reply:
Why not just select the datapoints closest to the run time range from each side? or even better only pull the relevant data from the database? e.g something like this:
```python
def before(dt):
    return [
    {
    "$match": {
            "time": {
                    "$lte": pd.to_datetime(dt),
                }
        }
    },
    
    {
    "$sort": {"time": 1}
    },
    {
    "$limit": 1
    }
]

def after(dt):
    return [
    {
    "$match": {
            "time": {
                    "$gte": pd.to_datetime(dt), 
                }
        }
    },
    
    {
    "$sort": {"time": -1}
    },
    {
    "$limit": 1
    }
]
def get_correction(db, correction_name, dt):
    before_df = pdm.read_mongo(correction_name, before(dt), db)
    after_df = pdm.read_mongo(correction_name, after(dt), db)
    return pd.concat([before_df, after_df])

```

Reply:
This is a great idea, I will implemented although this needs to be done at strax. I'm going to close this PR